System.Configuration.ConfigurationManager.Tests: use Assembly.Location to determine ThisApplicationPath.

tmds opened the pull request:
Fixes https://github.com/dotnet/runtime/issues/108476.

@ViktorHofer @steveharter ptal.

tmds committed:
System.Configuration.ConfigurationManager.Tests: use Assembly.Location to determine ThisApplicationPath.

jkotas commented on src/libraries/System.Configuration.ConfigurationManager/tests/Mono/TestUtil.cs:
```diff
@@ -58,8 +58,10 @@ public static string ThisApplicationPath
-                return Path.Combine(AppDomain.CurrentDomain.BaseDirectory,
-                    Assembly.GetEntryAssembly().ManifestModule.Name);
+                Assembly assembly = Assembly.GetEntryAssembly();
+                string directory = Path.GetDirectoryName(assembly.Location);
```
BaseDirectory is typically the directory of the entry assembly. Where is the difference in your test environment come from? Are you running the tests in some kind of atypical setup that makes BaseDirectory different from entry assembly?

jkotas replied:
This was done originally to make the test compatible with single file form factors. The description of the old PR that you have linked says that it is fixing the tests for uap-aot (aka .NET Native) that was the first single file form factor.

The problem with single file is that Assembly.Location returns empty string.

I think it would be best to delete the `ThisApplicationPath` property and the one test that uses it.  There are other more comprehensive tests [OpenExeConfiguration2](https://github.com/dotnet/runtime/blob/497236a2359f9de4b70b1dbd00e55449823e6f63/src/libraries/System.Configuration.ConfigurationManager/tests/Mono/ConfigurationManagerTest.cs#L79) and [OpenExeConfiguration2_ExePath_DoesNotExist](https://github.com/dotnet/runtime/blob/497236a2359f9de4b70b1dbd00e55449823e6f63/src/libraries/System.Configuration.ConfigurationManager/tests/Mono/ConfigurationManagerTest.cs#L107C21-L107C63) to exercise this scenario. These other tests create the files in known locations and so they do not depend on the deployment details that may change over time.

tmds replied:
Thanks for your insights! I've removed the test.

tmds committed:
Remove exePath_UserLevelNone test.

This is functionally covered other tests (OpenExeConfiguration2 and OpenExeConfiguration2_ExePath_DoesNotExist)
that do not rely on the test deployment.

jkotas commented:
/ba-g unrelated timeout
